CMakeLists.txt

```diff
@@ -59,6 +59,13 @@ endif ()
 # ignored.
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
+# Validate CMAKE_BUILD_TYPE
+string(TOUPPER "${CMAKE_BUILD_TYPE}" uppercase_CMAKE_BUILD_TYPE)
+list(APPEND uppercase_build_type "" "DEBUG" "RELEASE" "RELWITHDEBINFO" "MINSIZEREL")
```

Comment:
I don't see us actually testing `MINSIZEREL` so I'd suggest we remove that and just assert agains the other three.

Reply:
I took the basic types from CMAKE :). Indeed we don't use `MINSIZEREL` so I removed it.